packages/components/src/List/ListComposition/TextFilter/TextFilter.component.test.js

```diff
@@ -46,64 +46,68 @@ describe('TextFilter', () => {
 			textFilter: '',
 			setTextFilter: jest.fn(),
 		};
+		const event = { target: { value: 'my-filter-value' } };
 
 		// when
 		let wrapper;
 		act(() => {
 			wrapper = mount(
 				<ListContext.Provider value={context}>
-					<TextFilter id="myTextFilter" initialDocked />
+					<TextFilter id="myTextFilter" initialDocked debounceTimeout={0} />
 				</ListContext.Provider>,
 			);
 
-			wrapper
-				.find('FilterBar')
-				.at(0)
-				.prop('onToggle')();
-			wrapper
-				.find('FilterBar')
-				.at(0)
-				.prop('onFilter')({}, 'my-filter-value');
+			wrapper.find('button#myTextFilter').simulate('click');
 		});
 		wrapper.update();
+		wrapper.find('input').at(0).simulate('change', event);
 
 		// then
-		expect(
-			wrapper
-				.find('FilterBar')
-				.at(0)
-				.prop('docked'),
-		).toBe(false);
 		expect(context.setTextFilter).toHaveBeenCalledWith('my-filter-value');
 	});
 
-	it('should call the change callbacks when they are provided (controlled mode)', () => {
+	it('should call the toggle callback when they are provided (controlled mode)', () => {
```

Comment:
Split the tests into 2 tests:
- test toggle in controlled mode
- test change on controlled mode